# Homework 2

- Student name: Byron Washington
- Email: ********
- G-number: ********

## Imported packages

In [136]:
import math

## Question 1

### Question 1.a

False, python can is more of a general purpose programming language that can also be used for things like building apps or simply computing math problems

### Question 1.b

False, some operators have the same precedence (e.g. + and -), while other have different (e.g. + and *)

---

## Question 2

### Question 2.a

In [137]:
(9 + 0.2)/(10 + 15)

0.368

### Question 2.b

In [138]:
math.pow(2.2, -1) - math.pow(0.5, 0.5)

-0.25256132664109304

### Question 2.c

In [139]:
math.sqrt(81) - ((5) / abs(-3 * math.sqrt(abs(math.pow(1.1, 4) - 4.4))))

8.02730178221882

### Question 2.d

In [140]:
(2 / math.pi) * math.log10(math.e)

0.2764804542097449

---

## Question 3

In [141]:
intial_x = -10
v = 0.46 # 1.5 ft/s is 0.46 m/s
t = 5.5
final_x = intial_x + (v * t)
print('Final position is ' + str(final_x) + ' meters')

Final position is -7.47 meters


---

## Question 4

### Question 4.a

In [142]:
numerator = (1 * math.cos(math.pi) + math.sin(math.radians(60)))
denominator = 0.5
final_answer = numerator / denominator
print(final_answer)

-0.2679491924311228


### Question 4.b

In [143]:
numerator = math.atan(math.radians(75))
denominator = math.cos(math.radians(60)) + math.asin(math.radians(10))
final_answer = (numerator / denominator) * (math.pow(10, -1))
print(final_answer)

0.13597691875146167


### Question 4.c

In [144]:
final_answer = math.pow((((math.pi) / 2) * math.log10(4)), -2)
print(final_answer)

1.118100160770777


### Question 4.d

In [145]:
final_answer = math.pow(math.e, - math.sqrt(math.log(3, 6)))
print(final_answer)

0.4570158713547788


---

## Question 5

### Question 5.a

In [146]:
initial_y = 93.11
initial_v = 0
g = -9.8
t = 1.9
final_y_position = initial_y + (initial_v * t) + (0.5 * g * math.pow(t, 2))
print(str(final_y_position) + ' meters')

75.42099999999999 meters


The object does NOT reach the ground in 1.5 seconds because the final y position would be equal to 0, but with t = 1.9 the final y position is 75.4 meters

### Question 5.b

In [147]:
initial_y = 93.11
final_y = 0
initial_v = 0
g = -9.8
final_time1 = (-initial_v + math.sqrt(math.pow(initial_v, 2) - (2 * g * (initial_y - final_y)))) / g
final_time2 = (-initial_v - math.sqrt(math.pow(initial_v, 2) - (2 * g * (initial_y - final_y)))) / g
if final_time1 > 0:
	print(str(final_time1) + ' seconds')
else:
	print(str(final_time2) + ' seconds')

4.359133034942444 seconds
